tests/Mutator/Boolean/TrueValueTest.php

```diff
@@ -90,6 +91,133 @@ public function provideMutationCases(): array
 return false;
 PHP
                 ,
+            ];
+
+        yield 'It does not mutate when used in "in_array" function by default' => [
+                <<<'PHP'
+<?php
+
+in_array($a, $b, true);
+PHP
+                ,
+            ];
+
+        yield 'It does not mutate when used in "\in_array" function by default' => [
+                <<<'PHP'
+<?php
+
+\in_array($a, $b, true);
+PHP
+                ,
+            ];
+
+        yield 'It mutates when used in a method named "in_array"' => [
+                <<<'PHP'
+<?php
+
+$a->in_array($b, $c, true);
+PHP
+                ,
+            <<<'PHP'
+<?php
+
+$a->in_array($b, $c, false);
+PHP
+                ,
+            ];
+
+        yield 'It mutates when used in "\in_array" function and explicitly enabled in settings' => [
+                <<<'PHP'
+<?php
+
+\in_array($a, $b, true);
+PHP
+                ,
+            <<<'PHP'
+<?php
+
+\in_array($a, $b, false);
+PHP
+                ,
+                [
+                    'settings' => ['in_array' => true],
+                ],
+            ];
+
+        yield 'It does not mutate when used in "\in_array" function and explicitly disabled' => [
+                <<<'PHP'
+<?php
+
+\in_array($a, $b, true);
+PHP
+                ,
+                [],
+                [
+                    'settings' => ['in_array' => false],
+                ],
+            ];
+
+        yield 'It does not mutate when used in "array_search" function by default' => [
+            <<<'PHP'
+<?php
+
+array_search($a, $b, true);
+PHP
+            ,
+        ];
+
+        yield 'It does not mutate when used in "\array_search" function by default' => [
+            <<<'PHP'
+<?php
+
+\array_search($a, $b, true);
+PHP
+            ,
+        ];
+
+        yield 'It mutates when used in a method named "array_search"' => [
+            <<<'PHP'
+<?php
+
+$a->array_search($b, $c, true);
+PHP
+            ,
+            <<<'PHP'
+<?php
+
+$a->array_search($b, $c, false);
+PHP
+            ,
+        ];
+
+        yield 'It mutates when used in "\array_search" function and explicitly enabled in settings' => [
+            <<<'PHP'
+<?php
+
+\array_search($a, $b, true);
+PHP
+            ,
+            <<<'PHP'
+<?php
+
+\array_search($a, $b, false);
+PHP
+            ,
+            [
+                'settings' => ['array_search' => true],
+            ],
+        ];
+
+        yield 'It does not mutate when used in "\array_search" function and explicitly disabled' => [
```

Comment:
How about adding a test case where array_search is wrongly capitalized.

Reply:
done